pkg/types/apisix/v1/plugin_types.go

```diff
@@ -0,0 +1,50 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package v1
+
+// TrafficSplitConfig is the config of traffic-split plugin.
+// +k8s:deepcopy-gen=true
+type TrafficSplitConfig struct {
+	Rules []TrafficSplitConfigRule `json:"rules"`
+}
+
+// TrafficSplitConfigRule is the rule config in traffic-split plugin config.
+// +k8s:deepcopy-gen=true
+type TrafficSplitConfigRule struct {
+	WeightedUpstreams []TrafficSplitConfigRuleWeightedUpstream `json:"weighted_upstreams"`
+}
+
+// TrafficSplitConfigRuleWeightedUpstream is the weighted upstream config in
+// the traffic split plugin rule.
+// +k8s:deepcopy-gen=true
+type TrafficSplitConfigRuleWeightedUpstream struct {
+	UpstreamID string `json:"upstream_id,omitempty"`
+	Weight     int    `json:"weight"`
+}
+
+// IPRestrictConfig is the rule config for ip-restriction plugin.
+// +k8s:deepcopy-gen=true
+type IPRestrictConfig struct {
+	Whitelist []string `json:"whitelist,omitempty"`
```

Comment:
change all `Whitelist` to `Allowlist` ?

Reply:
This is for consistency as terms in APISIX are `black`, `white`.

Reply:
I'm not sure if APISIX will be modified, but these are two projects. I think it makes sense to modify it now. In order to avoid subsequent modifications caused by break changing.

Reply:
Good point.